### PR TITLE
Add filter questions

### LINKED ISSUE #53: F2.2.4 The service should allow filtering of questions by attributes like category, complexity.

### PR DESCRIPTION
**Pull request by wr1159**
Resolves #53 
Filter questions by complexity and category
- category currently is a search until we implement a schema / another end point where we can fetch the categories
- complexity is a select

### CODE REVIEW

File: frontend/components/questions/question-filter.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            <Label htmlFor="complexity">Complexity</Label>
+            <Select
+              value={complexity}
+              onValueChange={(value) => onComplexityChange(value)}
+            >
+              <SelectTrigger id="complexity" className="mt-1">
+                <SelectValue placeholder="Select complexity" />
+              </SelectTrigger>
+              <SelectContent>
+                <SelectItem value="all">All</SelectItem>
```
Comment (SelwynAng) [inline frontend/components/questions/question-filter.tsx:58]: May can use a complexity schema, and then map over these fields?

File: frontend/components/questions/question-filter.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "../ui/select";
```
Comment (SelwynAng) [inline frontend/components/questions/question-filter.tsx:14]: To use absolute import

**Review by SelwynAng — CHANGES_REQUESTED** [review]
Looks good to me overall but `yarn build` runs into error, similar to the one #129.

Screenshot 2024-09-26 at 12 11 50 PM

File: frontend/components/questions/question-filter.tsx (hunk 1 above)
Comment (wr1159) [reply frontend/components/questions/question-filter.tsx:58]: Using zod enum?

**Review by SelwynAng — APPROVED** [review]

**Pull request merged**